pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -1464,13 +1464,13 @@ func createTagQueryFromTagQueryParams(queryParams []model.TagQueryParam) []model
 	tags := []model.TagQuery{}
 	for _, tag := range queryParams {
 		if len(tag.StringValues) > 0 {
-			tags = append(tags, model.NewTagQueryString(tag.Key, tag.StringValues, tag.Operator))
+			tags = append(tags, model.NewTagQueryString(tag.Key, tag.StringValues, tag.Operator, tag.TagType))
 		}
 		if len(tag.NumberValues) > 0 {
-			tags = append(tags, model.NewTagQueryNumber(tag.Key, tag.NumberValues, tag.Operator))
+			tags = append(tags, model.NewTagQueryNumber(tag.Key, tag.NumberValues, tag.Operator, tag.TagType))
 		}
 		if len(tag.BoolValues) > 0 {
-			tags = append(tags, model.NewTagQueryBool(tag.Key, tag.BoolValues, tag.Operator))
+			tags = append(tags, model.NewTagQueryBool(tag.Key, tag.BoolValues, tag.Operator, tag.TagType))
```

Comment:
If we find ourselves passing all the existing fields on the struct, then it's better to pass the entire struct. The usual pattern is to use something like `Options` so any new option can be part of it i.e `model.New(options)`.